[Datasets] Update Ray Data documentation for lazy execution by default (1st part) (#32394)
ray-project/ray: doc/source/data/creating-datasets.rst

Deferred Read Task Execution
============================

> Datasets created via the ``ray.data.read_*()`` APIs are lazy: initially, none of
> read tasks will be executed. Metadata inspection functions like

Reviewer: ```suggestion
Datasets created via the ``ray.data.read_*()`` APIs are lazy: no read tasks are executed until a downstream consumption operation triggers execution. Metadata inspection functions like
```

Author: @clarkzinzow - updated.

:meth:`ds.schema() <ray.data.Dataset.schema>` and :meth:`ds.show() <ray.data.Dataset.show>`

> will trigger execution of only one or some tasks, instead of all tasks. This is enabling
> metadata can be inspected right away. Executing further transformations on the Dataset

Reviewer: ```suggestion
will trigger execution of only one or some tasks, instead of all tasks. This is allows
metadata to be inspected right away. Executing further transformations on the Dataset
```

> will trigger execution of only one or some tasks, instead of all tasks. This is enabling
> metadata can be inspected right away. Executing further transformations on the Dataset
> will trigger execution of all read tasks, and execution of all read tasks can be

Reviewer: ```suggestion
will trigger execution of only one or some tasks, instead of all tasks. This allows
metadata to be inspected right away. Execution of all read tasks can be
```

Reviewer: "Executing further transformations on the Dataset
will trigger execution of all read tasks," isn't true anymore now that transformations are lazy, and we already call out that downstream consumption operations trigger execution above with my suggested edit.

Author: @clarkzinzow - updated.
